packages/cli/lib/services/dryrun.service.ts

```diff
@@ -183,18 +183,40 @@ export class DryRunService {
             return;
         }
 
-        let normalizedInput;
         let stubbedMetadata;
-        try {
-            normalizedInput = JSON.parse(actionInput as unknown as string);
-        } catch {
-            normalizedInput = actionInput;
-        }
+        let normalizedInput;
 
-        try {
-            stubbedMetadata = JSON.parse(rawStubbedMetadata as unknown as string);
-        } catch {
-            stubbedMetadata = rawStubbedMetadata;
+        // the user can reference a file for input
+        // if it includes both '@' and 'json'
+        if (actionInput && actionInput.toString().includes('@') && actionInput.toString().includes('.json')) {
+            // fetch the file contents
+            const fileContents = localFileService.readFile(actionInput.toString());
+            if (!fileContents) {
+                console.log(chalk.red('The file could not be read. Please make sure it exists.'));
+                return;
+            }
+            normalizedInput = JSON.parse(fileContents);
```

Comment:
JSON.parse can throw an error, note that the error message is helpful but it could be better presented to the user